src/py42/exceptions.py

```diff
@@ -63,6 +63,44 @@ def __init__(self, error_message):
         super(Py42StorageSessionInitializationError, self).__init__(error_message)
 
 
+class Py42UserAlreadyAddedError(Py42Error):
+    """An exception raised when the user is already added to group or list, such as the
+    Departing Employee list."""
+
+    def __init__(self, user_id, list_name):
+        msg = u"User with ID {} is already on the {}.".format(user_id, list_name)
+        super(Py42UserAlreadyAddedError, self).__init__(msg)
+
+
+class Py42LegalHoldNotFoundOrPermissionDeniedError(Py42Error):
+    """An exception raised when a legal hold matter is inaccessible from your account."""
+
+    def __init__(self, matter_id):
+        super(Py42LegalHoldNotFoundOrPermissionDeniedError, self).__init__(
+            u"Matter with id={} either does not exist or your account does not have permission to view it.".format(
+                matter_id
+            )
+        )
+
+
+class Py42UserDoesNotExistError(Py42Error):
+    """An exception raised when a username is not in our system."""
+
+    def __init__(self, username):
+        super(Py42UserDoesNotExistError, self).__init__(
+            u"User '{}' does not exist.".format(username)
+        )
+
+
+class Py42InvalidRuleTypeError(Py42Error):
+    """An exception raised when trying to add or remove users to a system rule."""
+
+    def __init__(self, rule_id, source):
+        msg = u"Only alert rules with a source of 'Alerting' can be targeted by this command. "
+        msg += u"Rule {0} has a source of '{1}'."
+        super(Py42InvalidRuleTypeError, self).__init__(msg.format(rule_id, source))
```

Comment:
Can we have these be subclasses of Py42HTTPError and keep the original exception's response data on them?